final/implementation/frontends.tex

```diff
@@ -0,0 +1,121 @@
+\section{Frontends}
+\label{sec:frontends}
+
+\Cref{cha:design} showed that the backend places little restrictions on the
+frontends. This is illustrated by several facts, the first being that the
+evaluation pipelines that are constructed in the backend (see
+\cref{sec:function-comp}) are lazy. This allows clients to schedule and execute
+these pipelines on different threads. Secondly, the visitor pattern (see
+\cref{sec:visitor}) allows the frontends to process the results in many
+different ways.
+
+To illustrate this flexibility, two frontends have been developed that operate
+on two different spectra: a single threaded, text-based console frontend and a
+multithreaded plugin to the Eclipse IDE. Both of these are discussed next.
+
+\subsection{Console REPL}
+\label{ssec:consolerepl}
+
+% TODO: add note on singlehreading and blocking input?
```

Comment:
I think it is an important detail, especially to contrast the Eclipse plugin. Not sure where to add it, though. I'll look at it again tomorrow.